### PR TITLE
SIM-2556 Generator - Colocar webservice de healthcheck no generator

### PR DESCRIPTION
**Pull request by jplippi**

### CODE REVIEW

File: src/main/kotlin/org/usecase/app/Router.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import io.swagger.v3.oas.annotations.Operation
+import org.usecase.param.DefaultParam
+import org.usecase.user.context.PublicPipe
+import org.usecase.user.process.HealthCheckProcess
+import org.usecase.wrapper.RouterWrapper
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+
+@Path("/healthcheck")
+class Router : RouterWrapper() {
```
Comment (melanke) [inline src/main/kotlin/org/usecase/app/Router.kt:12]: Shouldn't this router be named "HealthCheckRouter"?

File: src/main/kotlin/org/usecase/app/Router.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    @GET
+    @Operation(tags = ["HealthCheck"], summary = "Checks status of DB connection and coroutines")
+    fun check() {
+        PublicPipe.apply {
+            handle(connectionPipe, DefaultParam()) { context ->
+                HealthCheckProcess(context).check()
+            }
+            handle(transactionPipe, DefaultParam()) { context ->
```
Comment (melanke) [inline src/main/kotlin/org/usecase/app/Router.kt:21]: why do we need this twice?

File: src/main/kotlin/org/usecase/dao/HealthCheckDao.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * Data Access Object of Health Check
+ * @author Simpli CLI generator
+ */
+class HealthCheckDao(val con: AbstractConnector) {
+    fun getDate(): Date? {
+        val query = Query().selectRaw("NOW()")
+
+        return con.getOne(query) {
```
Comment (melanke) [inline src/main/kotlin/org/usecase/dao/HealthCheckDao.kt:15]: you can use `con.getFirstDate(query)`

File: src/main/kotlin/org/usecase/dao/HealthCheckDao.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package org.usecase.dao
```
Comment (ftgibran) [inline src/main/kotlin/org/usecase/dao/HealthCheckDao.kt:1]: The problem of using this path is that all content inside it is created dynamically. So, if there is a table named health_check, its DAO will conflict with this file.

File: src/main/kotlin/org/usecase/user/process/HealthCheckProcess.kt [hunk 5]
```diff
@@ -0,0 +1,47 @@
+package org.usecase.user.process
```
Comment (ftgibran) [inline src/main/kotlin/org/usecase/user/process/HealthCheckProcess.kt:1]: The same idea above. I think it is better to place all these files (HealthCheckRouter, HealthCheckProcess, HealthCheckDao) inside `org.usecase.user.app.healthCheck`.

File: src/main/kotlin/org/usecase/app/Router.kt (hunk 2 above)
Comment (jplippi) [reply src/main/kotlin/org/usecase/app/Router.kt:21]: One for each pipe. Usecae uses the same datasource for different pipes, but often we use different data sources for writing and reading (read replica). I think it's better if it's already separate, that way just changing the datasource of the pipe will suffice

**Pull request merged**